riscv/csrs.cc

```diff
@@ -790,8 +790,9 @@ reg_t base_atp_csr_t::compute_new_satp(reg_t val) const noexcept {
   reg_t rv64_ppn_mask = (reg_t(1) << (MAX_PADDR_BITS - PGSHIFT)) - 1;
 
   reg_t mode_mask = proc->get_xlen() == 32 ? SATP32_MODE : SATP64_MODE;
+  reg_t asid_mask = proc->supports_impl(IMPL_MMU_ASID) ? proc->get_xlen() == 32 ? SATP32_ASID : SATP64_ASID : 0;
```

Comment:
It's a minor stylistic point, but I'd suggest breaking up this statement like the following (or whatever you prefer):

    reg_t asid_mask_if_mmu_enabled = proc->get_xlen() == 32 ? SATP32_ASID : SATP64_ASID;
    reg_t asid_mask = proc->supports_impl(IMPL_MMU_ASID) ? asid_mask_if_mmu_enabled : 0;